sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -120,6 +120,32 @@ object ScalaReflection extends ScalaReflection {
   }
 
   /**
+   * Returns the element type for Seq[_] and its subclass.
+   */
+  def getElementTypeForSeq(t: `Type`): Option[`Type`] = {
+    if (!(t <:< localTypeOf[Seq[_]])) {
+      return None
+    }
+    val TypeRef(_, _, elementTypeList) = t
```

Comment:
This would probably be better solved by using `match` and `typeParams`. Something like this:

```scala
t match {
  case TypeRef(_, _, Seq(elementType)) => Some(elementType)
  case _ =>
    t.baseClasses.find { c =>
      val cType = c.asClass.toType
      cType <:< localTypeOf[Seq[_]] && cType.typeParams.nonEmpty
    }.map(t.baseType(_).typeParams.head)
}
```

Also not sure whether types with more than one type parameter are handled correctly.